Utilities/AtomicCounter.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+class AtomicCounter {
+  private let lock = NSLock()
```

Comment:
Why does this need to have its own lock rather than having an `Atomic<Int>` for the value?

Reply:
That would only work if `Atomic<Int>` had a method that increments the value in one lock/unlock, as incrementing "manually" would imply that you get the original and set the incremented value - thus locking twice. I can add an extension that does just that. In that case I feel like I don't need this class at all.

My initial idea was to subclass `Atomic<Int>`, but that only worked with a recursive lock.

Reply:
ahhhh yeah because you need one lock to access and another to modify. Hmmmm....

I like the idea of not having to have a separate class to modify the value. I wonder if we could do something like a `updateValue(_ value: @autoclosure () -> T)`, which would allow access+modify with the built-in lock. I'm not sure that's totally possible with an overridden setter though...

Reply:
You can work with _value directly so it would be possible.
It would look like `updateValue(_ value: (T) -> T)` though, at least for this use case.

Reply:
`updateValue(_ value: (T) -> T) -> T` actually, which might be a bit convoluted :D

Reply:
I did it with a simple extension of the Atomic<Int> type, to keep it simple for now.